records/2022-03-08.priority-lanes.md

```diff
@@ -0,0 +1,71 @@
+# Enforcing priority lanes
+
+Date: 2022-03-08
+
+## Status
+
+**DRAFT.**
+
+## Context
+
+Listing contextual elements to consider prior to offer suggestions and deepen our understanding of current limitations.
+
+### Redis inboxes
+
+We currently have 2 inboxes for batch saving via Redis (email and sms). Batches of notifications are taken out in the order they were POSTed. This means high priority notifications might have to wait behind a large number of bulk notifications.
+
+## Current Implementation
+
+![Current Batch Saving](https://user-images.githubusercontent.com/8869623/160944266-48cf19fb-ad07-411b-8f73-88616697f9a8.jpeg)
+
+We store all notifications that come throught the API into a Redis List (Inbox). A process takes the notifications from Inbox and makes lists of a certain chunk size. Each list get processed as a single unit and gets saved to the Database. Instead of saving a single notification at a time, we are now saving a list of notifications (batch) in a single transacation.
```

Comment:
Hmm while the priority lane will get replicated at the batch level, it's also meant for the dedicated database save. So it's not all tied to the batch saving: we should have done this batch before batch saving implementation. 

The current implementation is currently honored through the various celery tasks and workers but not upstream (db save + batch saving levels). We need to adjust these upstreams.